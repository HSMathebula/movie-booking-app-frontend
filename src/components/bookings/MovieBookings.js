import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchBookings } from '../../redux/booking/bookings';
import { getCurrentUser } from '../../redux/authentication/authentication';

function MovieBookings() {
  const bookings = useSelector((state) => state.bookings.list);

  const user = getCurrentUser();
  const userId = user.user.id;

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchBookings(userId));
  }, []);

  return (
    <div className="my-booking-container">

      <div className="px-5 pb-2 text-center booking-heading mt-5">
        <h2>My bookings</h2>
      </div>

      {bookings.length === 0
        ? (
          <h2 className="not-found-notice mt-5 text-center">No booked movies</h2>
        )
        : (
          <div className="booking-box py-3">
            {bookings.map((booking) => (
              <div className="my-booking-card container" key={booking.bookingId}>
                <div className="d-flex ustify-content-evenly">
                  <div className="booking-container">
                    <h4 className="movie-title">{booking.movieTitle}</h4>
                    <img src={booking.moviePicture} className="card-img-top" alt={booking.title} />
                    <p className="booked-date">{booking.bookedDate}</p>
                    <p className="city-name">{booking.cityName}</p>
                  </div>
                </div>
              </div>
            ))}
          </div>
        )}
    </div>
  );
}

export default MovieBookings;
