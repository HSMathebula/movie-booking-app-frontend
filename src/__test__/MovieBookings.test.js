import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MovieBookings from '../components/bookings/MovieBookings';
import { Route } from 'react-router';

const user ={
    user: {
        id: 1,
        name: 'happen'
    }
}
const bookings = [
  {
    bookedDate: '2022-02-02',
    bookingId: 1,
    movieTitle: 'Star wars',
    cityName: 'Jozi',
    moviePicture:
      'https://image.shutterstock.com/image-vector/hawaii-beach-bar-retro-sign-600w-2156338567.jpg',
  },
];

describe('testing the rendering of the movie bookings page', () => {
  it('', () => {
    const { container } = render(
      <Provider store={store}>
        <MovieBookings bookings={bookings} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
