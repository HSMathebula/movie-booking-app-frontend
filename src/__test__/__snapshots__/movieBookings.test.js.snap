// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of the movie bookings page  1`] = `
<div>
  <div
    class="booking-container container"
  >
    <div
      class="px-5 pb-2 text-center booking-heading mt-5"
    >
      <h2
        class="my-bookings"
      >
        My Bookings
      </h2>
    </div>
    <h2
      class="not-found-notice mt-5 text-center"
    >
      No booked movies
    </h2>
  </div>
</div>
`;
